### PR TITLE
added executionResults to executeBlockEphemerally

### PR DESCRIPTION
**Pull request by fatemebagherii**

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -342,7 +357,34 @@ func ExecuteBlockEphemerally(
...
+		Bloom:       types.CreateBloom(receipts),
+		LogsHash:    rlpHash(ibs.Logs()),
+		Receipts:    receipts,
+		Rejected:    rejectedTxs,
+	}
+
+	return receipts, stateSyncReceipt, execRs, nil
+}
+
+type ExecutionResultFateme struct {
```
Comment (sudeepdino008) [inline core/blockchain.go]: rename this properly, and put near top of the file rather than in between

File: core/blockchain.go [hunk 2]
```diff
@@ -299,23 +314,23 @@ func ExecuteBlockEphemerally(
...
 		if bloom != header.Bloom {
-			return nil, nil, fmt.Errorf("bloom computed by execution: %x, in header: %x", bloom, header.Bloom)
+			return nil, nil, nil, fmt.Errorf("bloom computed by execution: %x, in header: %x", bloom, header.Bloom)
 		}
 	}
 	if !vmConfig.ReadOnly {
 		txs := block.Transactions()
 		if _, err := FinalizeBlockExecution(engine, stateReader, block.Header(), txs, block.Uncles(), stateWriter, chainConfig, ibs, receipts, epochReader, chainReader, false); err != nil {
-			return nil, nil, err
+			return nil, nil, nil, err //something that bothers me is these returns. It used to return when ever a trnasaction failed. So, maybe some of these returns are like that; can be replace with something else. Like we did with rejected transactions.
```
Comment (sudeepdino008) [inline core/blockchain.go]: we'll be adding a flag to collect the rejected txs or return right away. Let's not bother with it right now. add a "TODO" comment and let it be

File: core/blockchain.go [hunk 3]
```diff
@@ -230,7 +237,7 @@ func ExecuteBlockEphemerally(
 	epochReader consensus.EpochReader,
 	chainReader consensus.ChainHeaderReader,
 	contractHasTEVM func(codeHash common.Hash) (bool, error), // transpiled evm (splitting evm codes into even lower level instructions)
-) (types.Receipts, *types.ReceiptForStorage, error) {
+) (types.Receipts, *types.ReceiptForStorage, *ExecutionResultFateme, error) {
```
Comment (sudeepdino008) [inline core/blockchain.go]: have Receipts and REceiptForStorage inside the ExecutionREsult

File: core/blockchain.go [hunk 4]
```diff
@@ -342,7 +357,34 @@ func ExecuteBlockEphemerally(
...
+}
+
+type ExecutionResultFateme struct {
+	// StateRoot   common.Hash    `json:"stateRoot"` I think we don't need this
+	TxRoot      common.Hash    `json:"txRoot"`
+	ReceiptRoot common.Hash    `json:"receiptRoot"`
+	LogsHash    common.Hash    `json:"logsHash"`
+	Bloom       types.Bloom    `json:"logsBloom"        gencodec:"required"`
+	Receipts    types.Receipts `json:"receipts"`
+	Rejected    []*rejectedTx  `json:"rejected,omitempty"`
```
Comment (sudeepdino008) [inline core/blockchain.go]: needs to be Pascal case...as modules outside need to access it too

File: core/blockchain.go [hunk 5]
```diff
@@ -342,7 +357,34 @@ func ExecuteBlockEphemerally(
...
+}
+
+func rlpHash(x interface{}) (h common.Hash) {
```
Comment (sudeepdino008) [inline core/blockchain.go:386]: can we reused hashing.go:rlpHash function? rather than reimplementing this here?

**Review by sudeepdino008 — COMMENTED** [review]
none

File: core/blockchain.go (hunk 5 above)
Comment (fatemebagherii) [reply core/blockchain.go:386]: the `hashing.go:rlpHas` is not exported

File: core/blockchain.go (hunk 4 above)
Comment (fatemebagherii) [reply core/blockchain.go]: aren't they? I'm sorry, I don't understand

Comment (sudeepdino008) [reply core/blockchain.go]: i was talking about `rejectedTx`
it needs to be RejectedTx.

**Comment by sudeepdino008** [comment]
merged in ebe_migration branch - https://github.com/covalenthq/erigon/pull/6/commits

**Pull request closed without merging**